Tools/Scripts/webkitpy/w3c/test_importer.py

```diff
@@ -212,6 +212,14 @@ def __init__(self, port, test_paths, options):
     def do_import(self):
         if self.source_directory:
             source_path = str(Path(self.source_directory) / 'web-platform-tests')
+            if not Path(source_path).exists():
+                _log.error("There isn't a web-platform-tests directory in the path you have provided: %s" % source_path)
+                exit(1)
```

Comment:
Some tests fail because of the change of error message, and probably the exit. 
I could handle this differently like last time, with `try`, but I would like to be able to augment the tests. 

So I discovered `OutputCapture()`.

https://github.com/WebKit/WebKit/blob/e41ec2c29b8e53c5a5548c720195a808ac40124e/Tools/Scripts/libraries/webkitcorepy/webkitcorepy/output_capture.py#L72-L108

@JonWBedard @gsnedders is there a way to run the tests 

1. by targetting one test only. in pytest I'm used to do that instead of having to run the full testsuite all the time.
2. Making it more explicit about expected vs actual.

Excerpt when running `/usr/bin/python3 ./Tools/Scripts/test-webkitpy -v`
```
[1860/2635] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_import_dir_with_no_tests_and_no_hg erred:
  Traceback (most recent call last):
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 85, in test_import_dir_with_no_tests_and_no_hg
      importer.do_import()
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer.py", line 217, in do_import
      exit(1)
    File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/_sitebuiltins.py", line 26, in __call__
      raise SystemExit(code)
  SystemExit: 1

```

I understand well it doesn't mean the expectations, but the formatting is still a bit obscure. 
I guess there is a lot of history in this code.

Reply:
To run one test:

```
/usr/bin/python3 ./Tools/Scripts/test-webkitpy webkitpy.w3c.test_importer_unittest.TestImporterTest.test_import_dir_with_no_tests_and_no_hg
```

Reply:
ha… The solution is being caught I guess in the mock system of the test framework… hmm…

```
karlcow@Oison OpenSource % /usr/bin/python3 ./Tools/Scripts/test-webkitpy -v webkitpy.w3c.test_importer_unittest.TestImporterTest
Suppressing most webkitpy logging while running unit tests.
[1/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_non_dangling_platform passed
[2/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_import_dir_with_no_tests passed
[3/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_initpy_generation passed
[4/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_import_dir_with_no_tests_and_no_hg passed
[5/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_import_dir_with_empty_init_py erred:
  Traceback (most recent call last):
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 110, in test_import_dir_with_empty_init_py
      importer.do_import()
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer.py", line 218, in do_import
      sys.exit(1)
  SystemExit: 1
  
[6/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_checkout_directory passed
[7/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_clean_directory_option passed
[8/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_clean_directory_option_partial_import passed
[9/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_manual_slow_test passed            
[10/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_no_implicit_skip_with_download passed
[11/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_skip_test_import_download passed
[12/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_clean_directory_option_prefix_name passed
[13/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_remove_obsolete_content passed     
[14/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_template_test_variant passed
[15/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_template_test passed
[16/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_harnesslinks_conversion passed    
[17/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_template_test_variant_dangling passed
[18/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_update_slow_test passed           
[19/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_webkit_test_runner_options passed
[20/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_webkit_test_runner_import_reftests_with_absolute_paths_download passed
[21/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_keep_original_reftest_reference erred:
  Traceback (most recent call last):
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 565, in test_keep_original_reftest_reference
      fs = self.import_directory(['-s', '/home/user/wpt', '-d', '/mock-checkout/LayoutTests/w3c/web-platform-tests'], FAKE_FILES, ['web-platform-tests/css/css-images'])
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 125, in import_directory
      importer.do_import()
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer.py", line 218, in do_import
      sys.exit(1)
  SystemExit: 1
  
[22/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_webkit_test_runner_import_reftests_with_absolute_paths_from_source_dir erred:
  Traceback (most recent call last):
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 542, in test_webkit_test_runner_import_reftests_with_absolute_paths_from_source_dir
      fs = self.import_directory(['-s', '/home/user/wpt', '-d', '/mock-checkout/LayoutTests/w3c/web-platform-tests'], FAKE_FILES, ['web-platform-tests/css/css-images'])
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 125, in import_directory
      importer.do_import()
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer.py", line 218, in do_import
      sys.exit(1)
  SystemExit: 1
  
[23/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_no_implicit_skip_with_source erred:
  Traceback (most recent call last):
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 232, in test_no_implicit_skip_with_source
      fs = self.import_directory(['-s', '/home/user/wpt', '-d', '/mock-checkout/LayoutTests/w3c/web-platform-tests'], FAKE_FILES, ['web-platform-tests/css/css-images'])
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 125, in import_directory
      importer.do_import()
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer.py", line 218, in do_import
      sys.exit(1)
  SystemExit: 1
  
[24/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_skip_test_import_source erred:
  Traceback (most recent call last):
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 204, in test_skip_test_import_source
      fs = self.import_downloaded_tests(['-s', '/home/user/wpt', '--no-fetch', '-d', 'w3c'], FAKE_FILES)
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 143, in import_downloaded_tests
      importer.do_import()
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer.py", line 218, in do_import
      sys.exit(1)
  SystemExit: 1
  
[25/25] webkitpy.w3c.test_importer_unittest.TestImporterTest.test_crash_test_with_resource_file erred:
  Traceback (most recent call last):
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 467, in test_crash_test_with_resource_file
      fs = self.import_directory(['-s', '/home/user/wpt', '-d', '/mock-checkout/LayoutTests/imported/w3c/web-platform-tests'], FAKE_FILES, ['web-platform-tests/css/css-images'])
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer_unittest.py", line 125, in import_directory
      importer.do_import()
    File "/Users/karlcow/code/OpenSource/Tools/Scripts/webkitpy/w3c/test_importer.py", line 218, in do_import
      sys.exit(1)
  SystemExit: 1
  
Ran 25 tests in 1.213s
FAILED (failures=0, errors=6)
```

Reply:
It doesn't see the test files as it should.